gnat2goto/driver/tree_walk.adb

```diff
@@ -2882,6 +2916,24 @@ package body Tree_Walk is
 
       --  Begin processing for Do_Object_Declaration_Full_Declaration
 
+      procedure Register_Identifier_In_Symbol_Table (N : Irep; Val : Irep)
```

Comment:
There's nothing there now, but I remember a PR by Petr that was adding like a couple more similar functions. They should be refactored in the future into one, but not as part of this PR.

Reply:
This should probably go into the symbol table. I'm not sure about the arguments either; Rather than having an `Irep` and requiring ito to be a `Symbol_Expr`, why not pass in the `name` and `type` (which are ultimately what you need) as parameters?

Should also be called something like `Declare_Global_Variable`?

Also this should probably go into `goto_utils`

Reply:
Again, because a number of similar functions were introduced recently for very similar reasons, just in different paths, I suggest that this be done as a more organised attempt to refactor these functions and this one, and make sure that the refactored version gets appropriately invoked in the correct places. But I don't think it's within the scope of this PR for this to happen.